ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/transform/PubsubMessageToObjectNode.java

```diff
@@ -361,22 +362,31 @@ private void processField(String jsonFieldName, Field field, JsonNode value, Obj
             if (!Json.isNullOrEmpty(props)) {
               repeatedAdditionalProperties.add(props.get(jsonFieldName));
             } else {
-              repeatedAdditionalProperties.addNull();
+              repeatedAdditionalProperties.addObject();
             }
           }
         } else {
+          ArrayNode filteredValue = Json.createArrayNode();
           for (JsonNode record : value) {
             final ObjectNode props = additionalProperties == null ? null : Json.createObjectNode();
-            transformForBqSchema((ObjectNode) record, field.getSubFields(), props);
-            if (!Json.isNullOrEmpty(props)) {
-              repeatedAdditionalProperties.add(props);
+            if (record.isObject()) {
+              filteredValue.add(record);
+              transformForBqSchema((ObjectNode) record, field.getSubFields(), props);
+              if (!Json.isNullOrEmpty(props)) {
+                repeatedAdditionalProperties.add(props);
+              } else {
+                repeatedAdditionalProperties.addObject();
+              }
             } else {
+              // BigQuery cannot load null values into an array, so we insert an empty object
+              // instead.
```

Comment:
I think this comment should reflect that we have only tested that record is not an object
```suggestion
              // BigQuery only allows objects in this array, so we insert an empty object instead.
```